meet_balls_app/tests.py

```diff
@@ -67,3 +74,89 @@ def test_successful_logout(self, client, player):
+    def player_rating(self, player):
+        return PlayerRating.objects.create(ball_game="Basketball", player=player, rating=5)
+
+    @pytest.fixture
+    def court(self):
+        return Court.objects.create(
+            x=Decimal('11'),
+            y=Decimal('22'),
+            city='TEST_CITY',
+            neighborhood='TEST_NEIGHBORHOOD',
+            max_players=30)
+
+    @pytest.fixture
+    def court_ball_game(self, court):
+        return CourtBallGame.objects.create(court=court, ball_game="Basketball")
+
+    @pytest.fixture
+    def game_event(self, court, court_ball_game):
+        return GameEvent.objects.create(
+            id=20,
+            time=timezone.now(),
+            level_of_game=5,
+            min_number_of_players=3,
+            max_number_of_players=5,
+            court=court, ball_game="Basketball")
+
+    @pytest.fixture
+    def game_event_player(self, game_event, player):
+        return GameEventPlayer.objects.create(game_event=game_event, player=player, ball_responsible=False)
+
+    def test_profile_view_no_data(self, client, player):
+        url = reverse('profile', args=[player.user.id])
+        response = client.get(url)
+        assert response.status_code == 200
+        assert player.user.username in response.content.decode()
+        assert player.favorite_ball_game in response.content.decode()
+
+    def test_profile_view_with_data(self,
+                                    client,
+                                    player,
+                                    player_rating,
+                                    court,
+                                    court_ball_game,
+                                    game_event,
+                                    game_event_player,
+                                    ):
+
+        url = reverse('profile', args=[player.user.id])
+        response = client.get(url)
+        assert response.status_code == 200
+        assert player.user.username in response.content.decode()
+        assert player.favorite_ball_game in response.content.decode()
+        assert str(player_rating.rating) in response.content.decode()
+        assert str(game_event.ball_game) in response.content.decode()
+
+    def test_display_edit_profile_form(self, client, player):
+        url = reverse('edit profile', kwargs={'id': player.user.id})
+        response = client.get(url)
+        assert response.status_code == 200
+        assert 'Edit Player Profile' in response.content.decode()
+
+    def test_edit_profile_post_invalid(self, client, player):
+        url = reverse('edit profile', kwargs={'id': player.user.id})
+        data = {
+            'birth_date': 'invalid_date',
+            'favorite_ball_game': 'Basketball',
+        }
+        response = client.post(url, data)
+        assert response.status_code == 302
+        assert response.url == reverse('edit profile', kwargs={'id': player.user.id})
+
+    def test_edit_profile_post_valid(self, client, player):
```

Comment:
Don't we also want to ensure that player.BallGame and birth_date has been updated accordingly

Reply:
yep my bad, will be fixed